Add Edit button to collection resource page

awead opened the pull request:
If the user has edit access, show a button that links to the collection edit form. This follows the same UI layout and styling as the resource page for works.

Fixes #741

mtribone commented:
![screen shot 2017-04-19 at 11 01 50 am](https://cloud.githubusercontent.com/assets/4163828/25187235/6eae030a-24f0-11e7-8ad0-cb463d05dd42.png)

mtribone commented:
![screen shot 2017-04-19 at 11 01 31 am](https://cloud.githubusercontent.com/assets/4163828/25187244/76a885d0-24f0-11e7-969a-22ed32a81721.png)

awead committed:
Add Edit button to collection resource page

If the user has edit access, show a button that links to the collection
edit form. This follows the same UI layout and styling as the resource
page for works.
